src/lib/utils/exceptn.cpp

```diff
@@ -68,104 +69,131 @@ std::string to_string(ErrorType type)
    return "Unrecognized Botan error";
    }
 
-Exception::Exception(const std::string& msg) : m_msg(msg)
+namespace {
+
+/**
+* Simple formatter utility.
+*
+* Should be replaced with std::format once that's available
+* on all our supported compilers.
+*/
+template<typename... T>
+std::string simple_format(T... args)
+   {
+   std::ostringstream oss;
+   ((oss << args), ...);
+   return oss.str();
+   }
```

Comment:
Should we have that more prominently in (say) `stl_util.h`?

Reply:
I'm hoping to write a (very simple) variant of `std::format` that just handles `{}` with no flags, etc and promote that to `stl_util.h`, then we have a very easy conversion path when `std::format` is available. We could even start using it today on MSVC, AIUI:

```
#if feature_check_for_std_format
  using simple_format = std::format;
#else
  std::string simple_format(...) {}
#endif
```

but I guess in the meantime we could promote this version and start using it

Reply:
I started on this but it actually becomes quite wide ranging as we use `std::ostringstream` to format exception messages and the like in a number of spots. I'll open a second followup PR here.

Reply:
#3440